compiler/noirc_frontend/src/parser/parser/primitives.rs

```diff
@@ -96,7 +100,7 @@ pub(super) fn turbofish<'a>(
 }
 
 pub(super) fn variable() -> impl NoirParser<ExpressionKind> {
-    path().map(ExpressionKind::Variable)
+    path(super::parse_type()).map(ExpressionKind::Variable)
```

Comment:
Can any variable now have turbofish, e.g. `let x: u8 = 3; assert(x::<> == 3)`?

Reply:
Oh, this was also the case before this PR. There's also `variable_no_turbofish`. Or at least turbofish at the end of a variable was allowed before this PR and before we allowed it in the middle of a path (though I think it's ignored if it's not used as a function).

Reply:
Rust allows the above btw